packages/salesforcedx-vscode-core/test/vscode-integration/orgBrowser/metadataOutlineProvider.test.ts

```diff
@@ -351,6 +578,154 @@ describe('load org browser tree outline', () => {
+      {
+        label: 'DocumentA',
+        fullName: 'DocumentA',
+        type: NodeType.MetadataComponent
+      },
+      {
+        label: 'DocumentB',
+        fullName: 'DocumentB',
+        type: NodeType.MetadataComponent
+      }
+    ];
+    const getCmpsStub = stub(
+      MetadataOutlineProvider.prototype,
+      'getComponents'
+      ).returns(expected.map(n => n.fullName));
+
+    const metadataObject = {
+      xmlName: 'Document Folder',
+      directoryName: 'testDirectory',
+      suffix: 'cls',
+      inFolder: true,
+      metaFile: false,
+      label: 'Document Folder'
+    };
+    const documentFolderNode = new BrowserNode(
+      'Document folder',
+      NodeType.Folder,
+      'DocumentFolder',
+      metadataObject
+    );
+    const cmpsNodes = await metadataProvider.getChildren(documentFolderNode);
+    compareNodes(cmpsNodes, expected);
+    getCmpsStub.restore();
+  });
+
+  it('should load component nodes when folder within Email Templates type is selected', async () => {
+    const expected = [
+      {
+        label: 'TemplateA',
+        fullName: 'TemplateA',
+        type: NodeType.MetadataComponent
+      },
+      {
+        label: 'TemplateB',
+        fullName: 'TemplateB',
+        type: NodeType.MetadataComponent
+      }
+    ];
+    const getCmpsStub = stub(
+      MetadataOutlineProvider.prototype,
+      'getComponents'
+      ).returns(expected.map(n => n.fullName));
+
+    const metadataObject = {
+      xmlName: 'Email Template Folder',
+      directoryName: 'testDirectory',
+      suffix: 'cls',
+      inFolder: true,
+      metaFile: false,
+      label: 'Email Template Folder'
+    };
+    const emailFolderNode = new BrowserNode(
+      'Email Template Folder',
+      NodeType.Folder,
+      'EmailFolder',
+      metadataObject
+    );
+    const cmpsNodes = await metadataProvider.getChildren(emailFolderNode);
+    compareNodes(cmpsNodes, expected);
+    getCmpsStub.restore();
+  });
+
+  it('should load component nodes when folder within Reports type is selected', async () => {
+    const expected = [
+      {
+        label: 'ReportA',
+        fullName: 'ReportA',
+        type: NodeType.MetadataComponent
+      },
+      {
+        label: 'ReportB',
+        fullName: 'ReportB',
+        type: NodeType.MetadataComponent
+      }
+    ];
+    const getCmpsStub = stub(
+      MetadataOutlineProvider.prototype,
+      'getComponents'
+      ).returns(expected.map(n => n.fullName));
+
+    const metadataObject = {
+      xmlName: 'Report Folder',
+      directoryName: 'testDirectory',
+      suffix: 'cls',
+      inFolder: true,
+      metaFile: false,
+      label: 'Report Folder'
+    };
+    const reportFolderNode = new BrowserNode(
+      'Report folder',
+      NodeType.Folder,
+      'ReportFolder',
+      metadataObject
+    );
+    const cmpsNodes = await metadataProvider.getChildren(reportFolderNode);
+    compareNodes(cmpsNodes, expected);
+    getCmpsStub.restore();
```

Comment:
Ideally cleanup should be done as part of the test(suite) teardown or in try..finally so that if a test fails it still gets called. But if thats how all existing tests are done, it's fine for now.